widgetastic_manageiq/__init__.py

```diff
@@ -394,18 +394,20 @@ def _values_to_add(self, values):
         return list(set(values) - set(self.selected_text))
 
     def fill(self, values):
-        if set(values) == set(self.selected_text):
-            return False
+        if values:
+            before = []
+            after = []
+            for value in values:
+                switch = self.switch_by_text(value)
+                before.append(switch.read())
+                if switch.selected:
+                    switch.fill(False)
+                else:
+                    switch.fill(True)
+                after.append(switch.read())
+            return before != after
```

Comment:
This could be optimized a bit more I think, but there are other considerations for how this should be implemented to fill.

Basically, you could have a sentinel boolean indicating a change, and update that sentinel when before != after. Then you don't have to compile or compare the whole list. Once sentinel is set, it won't get unset, so you can short-circuit the check of before/after anyway and skip the second read.

```
updated = False
for value in values:
    switch = self.switch_by_text(value)
    before = switch.read()
    switch.fill(not switch.selected)
    if not updated and before != switch.read():
        updated = True  # once set, the statement above will short-circuit
return updated
```